Sources/Sentry/SentryUIEventTracker.m

```diff
@@ -0,0 +1,166 @@
+
+#if SENTRY_HAS_UIKIT
+
+- (instancetype)initWithSwizzleWrapper:(SentrySwizzleWrapper *)swizzleWrapper
+                  dispatchQueueWrapper:(SentryDispatchQueueWrapper *)dispatchQueueWrapper
+                           idleTimeout:(NSTimeInterval)idleTimeout
+{
+    if (self = [super init]) {
+        self.swizzleWrapper = swizzleWrapper;
+        self.dispatchQueueWrapper = dispatchQueueWrapper;
+        self.idleTimeout = idleTimeout;
+    }
+    return self;
+}
+
+- (void)start
+{
+    [self.swizzleWrapper
+        swizzleSendAction:^(NSString *action, id target, id sender, UIEvent *event) {
+            [SentrySDK.currentHub.scope useSpan:^(id<SentrySpan> _Nullable span) {
+                if (target == nil || sender == nil) {
+                    return;
+                }
+
+                NSString *transactionName = [self getTransactionName:action target:target];
+
+                BOOL sameAction = [self.activeTransaction.name isEqualToString:transactionName];
+                if (sameAction) {
+                    [self.activeTransaction dispatchIdleTimeout];
+                    return;
+                }
+
+                [self.activeTransaction finish];
+
+                NSString *operation = [self getOperation:sender];
+
+                SentryTransactionContext *context =
+                    [[SentryTransactionContext alloc] initWithName:transactionName
+                                                         operation:operation];
+
+                BOOL ongoingScreenLoadTransaction = span != nil &&
+                    [span.context.operation isEqualToString:SentrySpanOperationUILoad];
+                BOOL ongoingManualTransaction = span != nil
+                    && ![span.context.operation isEqualToString:SentrySpanOperationUILoad]
+                    && ![span.context.operation containsString:SentrySpanOperationUIAction];
+
+                BOOL bindToScope = !ongoingScreenLoadTransaction && !ongoingManualTransaction;
+                SentryTracer *transaction =
+                    [SentrySDK.currentHub startTransactionWithContext:context
+                                                          bindToScope:bindToScope
+                                                customSamplingContext:@{}
+                                                          idleTimeout:self.idleTimeout
+                                                 dispatchQueueWrapper:self.dispatchQueueWrapper];
+
+                if ([[sender class] isSubclassOfClass:[UIView class]]) {
+                    UIView *view = sender;
+                    if (view.accessibilityIdentifier) {
+                        [transaction setTagValue:view.accessibilityIdentifier
+                                          forKey:@"accessibilityIdentifier"];
+                    }
+                }
+
+                transaction.finishCallback = ^(void) { self.activeTransaction = nil; };
+
+                self.activeTransaction = transaction;
```

Comment:
This is trick to explain.

 line 65 may not finish the span immediately, and the span keeps the callback attributed on line 95.

Then we change the activeTransaction, but when the previous one finishes, it will clean the activeTransacation, allowing us to create a third one similar to the second without extending the time of the second one.

Reply:
Excellent catch, thanks.